NotificationServiceExtension/NotificationService.swift

```diff
@@ -2,57 +2,178 @@
+                let handledAsTalkPage = self.handleTalkPageTypeNotificationsIfNeeded(notifications: finalNotifications, bestAttemptContent: bestAttemptContent, contentHandler: contentHandler)
+                
+                guard !handledAsTalkPage else {
+                    return
+                }
+                
+                //generic handling for other types
+                guard finalNotifications.count == 1 else {
+                    contentHandler(bestAttemptContent)
+                    return
+                }
+                
+                guard let pushContentText = finalNotifications.first?.pushContentText else {
+                    contentHandler(bestAttemptContent)
+                    return
+                }
+                
+                bestAttemptContent.body = pushContentText
+                contentHandler(bestAttemptContent)
+            }
+        }
     }
     
     override func serviceExtensionTimeWillExpire() {
         // Called just before the extension will be terminated by the system.
         // Use this as an opportunity to deliver your "best attempt" at modified content, otherwise the original push payload will be used.
-        //TODO: We will modify content in all error cases to "New activity on Wikipedia" (see: https://phabricator.wikimedia.org/T288773 > Push Notification Content section)
         if let contentHandler = contentHandler,
            let bestAttemptContent =  bestAttemptContent {
             contentHandler(bestAttemptContent)
         }
     }
+    
+    private func handleTalkPageTypeNotificationsIfNeeded(notifications: Set<RemoteNotificationsAPIController.NotificationsResult.Notification>, bestAttemptContent: UNMutableNotificationContent, contentHandler: ((UNNotificationContent) -> Void)) -> Bool {
+        
+        if allNotificationsAreForSameTalkPage(notifications: notifications) {
+            let messageText = String.localizedStringWithFormat(WMFLocalizedString("notifications-push-talk-messages-format", value: "{{PLURAL:%1$d|message|messages}}", comment: "Plural messages text to be inserted into push notification content - %1$d is replaced with the number of talk page messages."), notifications.count)
+            bestAttemptContent.subtitle = String.localizedStringWithFormat(WMFLocalizedString("notifications-push-talk-title-format", value: "New %1$@", comment: "Title text for a push notification that represents talk page messages - %1$@ is replaced with \"Messages\" text (can be plural or singular)."), messageText)
+            
+            if notifications.count == 1 {
+                guard let pushContentText = notifications.first?.pushContentText else {
+                    contentHandler(bestAttemptContent)
+                    return true
+                }
+                
+                bestAttemptContent.body = pushContentText
+                contentHandler(bestAttemptContent)
+                return true
```

Comment:
```suggestion
            if notifications.count == 1 {
                if let pushContentText = notifications.first?.pushContentText else {
                    bestAttemptContent.body = pushContentText
                }

                contentHandler(bestAttemptContent)
                return true
```

Reply:
Only one return?

Reply:
@mcleinman this should be done, I reworked it a bit with the refactor.